### PR TITLE
[5773] Requests to API 2 /audit/:id as developer, returns HTTP 403 instead of 200

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/5773

### CODE REVIEW

File: ui/app/src/components/AuditGrid/AuditGridUI.tsx [hunk 1]
```diff
@@ -371,7 +371,7 @@ export function AuditGridUI(props: AuditGridUIProps) {
         sortingOrder={['desc', 'asc']}
         sortModel={sortModel}
         sortingMode="server"
-        autoHeight
+        autoHeight={Boolean(auditLogs.length)}
```
Comment (jvega190) [inline ui/app/src/components/AuditGrid/AuditGridUI.tsx:374]: There was a visual issue when there were no logs (and Empty State shows) and the autoHeight was true. When there are rows the autoHeight works properly.

File: ui/app/src/components/AuditManagement/AuditManagement.tsx [hunk 2]
```diff
@@ -132,7 +132,7 @@ export function AuditManagement(props: AuditManagementProps) {
       setDialogParams(parametersLookup[id]);
       auditLogEntryParametersDialogState.onOpen();
     } else {
-      fetchAuditLogEntry(id).subscribe({
+      fetchAuditLogEntry(id, site).subscribe({
```
Comment (rart) [inline ui/app/src/components/AuditManagement/AuditManagement.tsx:135]: What about global?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/AuditManagement/AuditManagement.tsx (hunk 2 above)
Comment (jvega190) [reply ui/app/src/components/AuditManagement/AuditManagement.tsx:135]: Site prop is optional (null in global), and the service validates siteId, if null, the param is not sent

File: ui/app/src/services/audit.ts [hunk 3]
```diff
@@ -59,8 +59,8 @@ export function fetchAuditLog(options: AuditOptions): Observable<PagedArray<Audi
   );
 }
 
-export function fetchAuditLogEntry(id: number): Observable<AuditLogEntry> {
-  return get<Api2ResponseFormat<{ auditLog: AuditLogEntry }>>(`/studio/api/2/audit/${id}`).pipe(
-    pluck('response', 'auditLog')
-  );
+export function fetchAuditLogEntry(id: number, siteId?: string): Observable<AuditLogEntry> {
+  return get<Api2ResponseFormat<{ auditLog: AuditLogEntry }>>(
+    `/studio/api/2/audit/${id}${siteId ? `?siteId=${siteId}` : ''}`
```
Comment (rart) [inline ui/app/src/services/audit.ts]: use `toQueryString` util

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/services/audit.ts [hunk 4]
```diff
@@ -59,8 +59,9 @@ export function fetchAuditLog(options: AuditOptions): Observable<PagedArray<Audi
   );
 }
 
-export function fetchAuditLogEntry(id: number): Observable<AuditLogEntry> {
-  return get<Api2ResponseFormat<{ auditLog: AuditLogEntry }>>(`/studio/api/2/audit/${id}`).pipe(
+export function fetchAuditLogEntry(id: number, siteId?: string): Observable<AuditLogEntry> {
+  const qs = toQueryString({ siteId });
+  return get<Api2ResponseFormat<{ auditLog: AuditLogEntry }>>(`/studio/api/2/audit/${id}${qs}`).pipe(
```
Comment (rart) [inline ui/app/src/services/audit.ts]: Did you test this?? Without the `skipNull` siteId argument will be sent empty. Global shouldn't have a siteId
```suggestion
  const qs = toQueryString({ siteId }, { skipNull: true });
  return get<Api2ResponseFormat<{ auditLog: AuditLogEntry }>>(`/studio/api/2/audit/${id}${qs}`).pipe(
```

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/services/audit.ts (hunk 4 above)
Comment (jvega190) [reply ui/app/src/services/audit.ts]: I did, when null the queryString ends up like `?`, no siteId parameter

**Review by rart — APPROVED** [review]

**Pull request merged**